server/src/main/java/server/bazel/bazelWorkspaceAPI/WorkspaceAPI.java

```diff
@@ -0,0 +1,213 @@
+package server.bazel.bazelWorkspaceAPI;
+
+import server.bazel.tree.SourceFile;
+import server.bazel.tree.WorkspaceTree;
+import server.bazel.tree.BuildTarget;
+import server.bazel.tree.WorkspaceTree.Node;
+import server.bazel.tree.Package;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class WorkspaceAPI {
+    private WorkspaceTree workspaceTree;
+
+    public WorkspaceAPI(WorkspaceTree workspaceTree) throws WorkspaceAPIException {
+        if (workspaceTree.getRoot() == null){
+            throw new WorkspaceAPIException("Workspace root cannot be null");
+        }
+        this.workspaceTree = workspaceTree;
+    }
+
+    /**
+     *
+     * @param workspaceTree an initialized workspace Tree object
+     * @return The current WorkspaceAPI object with the newly set workspaceTree
+     * @throws WorkspaceAPIException if WorkspaceTree has no root
+     */
+    public WorkspaceAPI setWorkspace(WorkspaceTree workspaceTree) throws WorkspaceAPIException {
+        if (workspaceTree.getRoot() == null){
+            throw new WorkspaceAPIException("Workspace root cannot be null");
+        }
+        this.workspaceTree = workspaceTree;
+        return this;
+    }
+
+    // Publicly available methods
+    /**
+     *
+     * @param currentPath The pathway represented as a String to the package where you want to look for other possible packages
+     *          expected format "//path/to/"
+     * @return A list of possible paths as Strings
+     *          expected output: list = {//path/to/available, ...}
+     * @throws WorkspaceAPIException if path is invalid
+     */
+    public List<String> findPossibleCompletionsForPath(String currentPath) throws WorkspaceAPIException {
+        ArrayList<String> allPossiblePaths = new ArrayList<>();
+        // This will throw an exception if the path is invalid
+        List<Package> allPossiblePackages = findNodeOfGivenPackagePath(PathType.PACKAGE_PATH, currentPath).getAllPackagesOfChildren();
```

Comment:
Looking at this now, we might only want the possibly completions logic in the `completion` package, but I think this is great for now.